### ISSUE TITLE
Significant performance gap on FashionIQ val set compared to reported results in Table 1

### ISSUE THREAD

**Issue by SunQian-Qian** [Chen-Junyang-cn/PLI#3]
Hi, thank you for your great work!

I tried to reproduce the results using the following command:
```python
CUDA_VISIBLE_DEVICES=0 python main.py \
--mask_ratio 0.75 \
--img_weight 0.25 \
--clip_image_model "ViT-L/14" \
--clip_text_model "ViT-L/14" \
--batch_size 64 \
--experiment_description "FixCFBug_mask_pretrain_clip_L_ImageNet" \
--compositor "None" \
--tuning_module both \
--epoch 10 \
--lr 1e-6 \
--pretrain_dataset ImageNet \
--trainer pretrainer \
--evaluator_code pretrainer_evaluator \
--image_encoders clip_image_encoder \
--text_encoders clip_text_encoder \
--config_path config/fashionIQ_CLIP_config.json```



I used the ImageNet-1k dataset for pretraining, following the exact setup described in the paper (same construction method), and I did not modify any code.

However, the validation results on the FashionIQ dataset are significantly lower than those reported in Table 1 of the paper:
	•	Shirt: R@10 = 30.37, R@50 = 48.48
	•	Dress: R@10 = 23.699, R@50 = 43.679
	•	Toptee: R@10 = 31.056, R@50 = 51.963

These are quite far from the reported numbers. Could you please help me understand what might be causing this discrepancy? Is there any additional step or configuration I might be missing?

Thanks in advance!

**Comment by Chen-Junyang-cn** [comment]
If you are using ImageNet-1k, please confirm: 1. Whether the split of your dataset is consistent with our paper. 2. The BLIP model needs to be used to generate the caption.

In addition, we recommend LLaVA-CC3M-Pretrain-595K for training the model, which will achieve better results.

Finally, you can set smaller epochs and observe the performance on each epoch to avoid overfitting.

**Comment by SunQian-Qian** [comment]
Hi, thank you very much for your helpful reply!

For the experiments based on ImageNet-1k pretraining:
	•	The dataset split is consistent with your paper (10K test split).
	•	The caption generation model is also consistent (BLIP2-FlanT5-XL).
	•	I set the number of epochs to 10, with the best results achieved around the third epoch.

I have also tried training with the LLaVA-CC3M-Pretrain-595K checkpoint as you suggested. The results improved slightly, but there is still a noticeable gap compared to the numbers reported in the paper:
	•	Shirt: R@10 = 32.336, R@50 = 49.804
	•	Dress: R@10 = 24.492, R@50 = 45.662
	•	Toptee: R@10 = 33.962, R@50 = 54.513

Could you kindly advise what other factors might be causing this gap?
Also, would it be possible for you to share some pretrained checkpoints for reference?

Additionally, the validation set was prepared following the data/fashionIQ.py script, with only minor adjustments to the directory structure and file names (no changes to the dataset content itself).

While working with the validation dataset, I noticed that in the FashionIQTestQueryDataset and FashionIQTestDataset classes, the code uses the `_get_img_caption_txt `function to load caption data, referring to .txt files located at:

```python
def _get_img_caption_txt(dataset_root, clothing_type, split):
    with open(os.path.join(dataset_root, 'captions_pairs', 'fashion_iq-{}-cap-{}.txt'.format(split, clothing_type))) as f:
        file_content = f.readlines()
    return file_content
```

However, these `.txt `files were not present in the original FashionIQ dataset.
Since the `_get_img_caption_json` function is already defined in the provided codebase, I used it instead to load the available `.json` files:
```python
def _get_img_caption_json(dataset_root, clothing_type, split):
    with open(os.path.join(dataset_root, 'captions', 'cap.{}.{}.json'.format(clothing_type, split))) as json_file:
        img_caption_data = json.load(json_file)
    return img_caption_data
```
May I ask if you performed any additional processing to generate the missing `.txt` files during your experiments?
If so, would it be possible to share these files or the preprocessing steps you applied?


Thanks again for your time and support!

**Comment by Chen-Junyang-cn** [comment]
Thank you for pointing out the problem. We have supplemented FashionIQ with txt and json data. We will also upload model checkpoints.

Finally, our code is modified from https://github.com/Monoxide-Chen/uncertainty_retrieval. Refer to this repository for possible dataset preprocessing.

**Comment by SunQian-Qian** [comment]
Hi, thank you very much for your kind and informative reply!

I appreciate you uploading the .txt and .json files for FashionIQ.

After reviewing the setup again, I believe I’ve identified the source of the performance gap: it seems that your evaluation follows the VAL-Split method as described in [Chen et al., CVPR 2020](https://openaccess.thecvf.com/content_CVPR_2020/papers/Chen_Image_Search_With_Text_Feedback_by_Visiolinguistic_Attention_Learning_CVPR_2020_paper.pdf), whereas I was using the Original-Split proposed in the [FashionIQ paper (Wu et al., CVPR 2021)](https://openaccess.thecvf.com/content/CVPR2021/papers/Wu_Fashion_IQ_A_New_Dataset_Towards_Retrieving_Images_by_Natural_CVPR_2021_paper.pdf).

The Original-Split contains a larger and more diverse candidate pool with greater distribution complexity, making the retrieval task more challenging. As a result, it generally yields lower performance compared to the VAL-Split setup.

Thanks again for your support and clarification — it has been extremely helpful in aligning my evaluation protocol more closely with yours.

**Comment by Chen-Junyang-cn** [comment]
We're glad our response was helpful.  You're completely right — the performance difference likely comes from the distinct evaluation splits used (VAL-Split vs. Original-Split).  We followed the VAL-Split setup in our work, as it allows for more controlled and reproducible comparisons, but we fully agree that the Original-Split presents a more challenging and realistic retrieval scenario.

To support further experimentation on different settings, we’ve also released the model checkpoints so that you (and others) can easily evaluate under various protocols, including the Original-Split.  We hope this helps enable fair and flexible comparisons across setups.
